contracts/sol6/KyberHintHandler.sol

```diff
@@ -406,6 +406,10 @@ abstract contract KyberHintHandler is IKyberHint, Utils5 {
         bytes32[] memory reserveIds,
         uint256[] memory splits
     ) internal pure returns (HintErrors) {
+        if (tradeType == TradeType.BestOfAll) {
+            if (reserveIds.length != 0 || splits.length != 0) return HintErrors.NonEmptyDataError;
```

Comment:
`splits != 0` is checked in line 428, will throw the `HintErrors.SplitsNotEmptyError`.... So maybe we just need to check `reserveIds.length != 0`?

Reply:
lets continue with this version

we supply good enough error messages.
late for another change.

Reply:
I think this is more clear error type for BestOfAll case